File: core/src/main/java/forpleuvoir/dhwu1a/core/event/group/BotLeaveEventActive.java
package forpleuvoir.dhwu1a.core.event.group;

import com.google.gson.annotations.SerializedName;
import forpleuvoir.dhwu1a.core.common.data.GroupData;
import forpleuvoir.dhwu1a.core.user.Group;

import javax.annotation.Nullable;

import static forpleuvoir.dhwu1a.core.common.ApiKey.GROUP;

/**
 * Bot主动退出一个群
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.group
 * <p>#class_name BotLeaveEventActive
 * <p>#create_time 2021/7/3 23:05
 */
public class BotLeaveEventActive extends GroupEvent {

    /**
     * Bot退出的群的信息
     */
    @SerializedName(GROUP)
    public final GroupData group;

    private BotLeaveEventActive(GroupData group) {
        super(GroupEventType.BotLeaveEventActive);
        this.group = group;
    }

    /**
     * 退出群之后则不再能获取到该群对象
     *
     * @return null
     */
    @Nullable
    @Override
    public Group getGroup() {
        return null;
    }

    @Override
    public String toPlainText() {
        return String.format("%s[group:%d]", type, group.id);
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/group/BotLeaveEventActive.kt
package forpleuvoir.dhwu1a.core.event.group

import com.google.gson.annotations.SerializedName
import forpleuvoir.dhwu1a.core.common.GROUP
import forpleuvoir.dhwu1a.core.common.data.GroupData
import forpleuvoir.dhwu1a.core.user.Group

/**
 * Bot主动退出一个群
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.group
 *
 * #class_name BotLeaveEventActive
 *
 * #create_time 2021/7/3 23:05
 */
class BotLeaveEventActive private constructor(group: GroupData) : GroupEvent(GroupEventType.BotLeaveEventActive) {
    /**
     * Bot退出的群的信息
     */
    @SerializedName(GROUP)
    val group: GroupData

    /**
     * 退出群之后则不再能获取到该群对象
     *
     * @return null
     */
    override fun getGroup(): Group? {
        return null
    }

    override fun toPlainText(): String {
        return String.format("%s[group:%d]", type, group.id)
    }

    init {
        this.group = group
    }
}